src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerHealthMonitor.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.CosmosDB
+{
+    internal class CosmosDBTriggerHealthMonitor
+    {
+        private readonly ILogger logger;
+
+        public CosmosDBTriggerHealthMonitor(ILogger logger)
+        {
+            this.logger = logger;
+        }
+
+        public Task OnErrorAsync(string leaseToken, Exception exception)
+        {
+            switch (exception)
+            {
+                case ChangeFeedProcessorUserException userException:
+                    this.logger.LogWarning(userException.InnerException, "Lease {LeaseToken} encountered an unhandled user exception during processing.", leaseToken);
+                    this.logger.LogDebug("Lease {LeaseToken} has error diagnostics {Diagnostics}", leaseToken, userException.ChangeFeedProcessorContext.Diagnostics);
```

Comment:
I'd suggest that you assign an EventId to give a better description for all of these new messages.